crates/agglayer-certificate-orchestrator/src/lib.rs

```diff
@@ -573,7 +586,68 @@ where
                     }
                 }
             }
-            _ => {}
+            Err(error) => {
+                warn!("Error during certification process: {}", error);
+                let certificate_error: Option<(CertificateId, CertificateStatusError)> = match error
+                {
+                    Error::ProofVerificationFailed {
+                        source,
+                        certificate_id,
+                    } => Some((certificate_id, source.into())),
+
+                    Error::ProverExecutionFailed {
+                        source,
+                        certificate_id,
+                    } => Some((
+                        certificate_id,
+                        CertificateStatusError::ProofGenerationError {
+                            generation_type: agglayer_types::GenerationType::Prover,
+                            source,
+                        },
+                    )),
+
+                    Error::NativeExecutionFailed {
+                        source,
+                        certificate_id,
+                    } => Some((
+                        certificate_id,
+                        CertificateStatusError::ProofGenerationError {
+                            generation_type: agglayer_types::GenerationType::Native,
+                            source,
+                        },
+                    )),
+
+                    Error::Types {
+                        source,
+                        certificate_id,
+                    } => Some((certificate_id, source.into())),
+
+                    Error::Storage(error) => {
+                        warn!(
+                            "Storage error happened in the certification process: {:?}",
+                            error
+                        );
+                        None
+                    }
+                    _ => None,
+                };
+
+                if let Some((certificate_id, error)) = certificate_error {
+                    if self
+                        .state_store
+                        .update_certificate_header_status(
+                            &certificate_id,
+                            &CertificateStatus::InError { error },
+                        )
+                        .is_err()
```

Comment:
Are the error-ed out certificate headers cleaned up from storage at some point or do we keep them forever? Seems like a possible DoS vector, at least until some sort of rate limiting is in place.